tox.ini

```diff
@@ -9,17 +9,38 @@ deps =
     flake8    
     isort    
     pytest
+    pytest-cov
 
 commands =    
-    black --check ccn-template
-    isort --check ccn-template
-    flake8 ccn-template --max-complexity 10
-    #pytest tests/
-    #coverage run --source=ccn-template --branch -m pytest tests/
-    #coverage report -m
+    black --check src
+    isort --check src
+    flake8 --config={toxinidir}/tox.ini src
+    pytest
 
 [gh-actions]
 python =
     3.8: py38
     3.9: py39
     3.10: py310    
```

Comment:
for the first point, it doesn't run on all python automatically: it is just configured to be able to run on all those. it just runs on what you specify on the github action, the -e py runs whatever the action specifies or whatever you have installed. py stays for any python version, the version has to be one of that specified in [gh-action]. test are run in parallel if you call the action with matrix as usual

Reply:
for the second point, that's the standard config of tox, but it is overwritten by the workflow if you set for example:
```
tox:
    strategy:
      matrix:
        os: [ubuntu-latest]
        python-version: ['3.9', '3.10']
```   
 this will run in parallel on 3.9 and 3.8 on

Reply:
so if you run tox locally, without specifying the python, it should run on all the versions specified in the ini

Reply:
Similar to `gen_ref_pages.py`, I think it would be helpful to have a simple motivating discussion of this in docs/notes/07. E.g., based on Billy's overview of github actions, my (total beginner) understanding is we bascially need some yml files to set up a build matrix and run some tests. What does `tox` give us over and above the yml machinery?

So I don't have any substantive comments just that I think it would be helpful to have a motivating bit about what we are doing here. :smile:

Reply:
I provided context that explains the advantages of using tox in a workflow